test/date_descriptor.test.js

```diff
@@ -301,6 +301,54 @@ describe("Valid narratives", () => {
   }
 })
 
+const redirects = [
+  ["WNV", "WNV/NA"],
+  /* Note that WNV is not in the index, so this versioned test ensures we are
+  not checking the existence/validity of any version descriptor against the WNV
+  resource, rather we are redirecting and deferring those checks */
+  ["WNV@2020-01-01", "WNV/NA@2020-01-01"],
+]
+
+describe("Paths redirect with version descriptors", () => {
+  /* See <https://github.com/node-fetch/node-fetch#manual-redirect> for how
+  fetch stores the redirect location when {redirect: 'manual'} */
+  
+  redirects.forEach(([fromUrl, toUrl]) => {
+    /* test RESTful API */
+    (['html', 'json']).forEach((type) => {
+      it(`REST API: ${fromUrl} goes to ${toUrl} (${type})`, async () => {
+        const res = await fetchType(`${BASE_URL}/${fromUrl}`, type);
+        expect(res.status).toEqual(307);
+        expect(res.headers.get('location')).toEqual(`${BASE_URL}/${toUrl}`);
+      })
+    })
+
+
+    /* test Charon API */
+    const charonUrl = (path, sidecar) =>
+      `${BASE_URL}/charon/getDataset?prefix=${path}${sidecar ? `&type=root-sequence` : ''}`;
+
+    /* note that the redirect URL of charon requests is constructed by creating
+    a URL object (the RESTful API does not) which results in '/' and '@' being
+    escaped in the query. This isn't necessary, but it's also fine to do so. */
+
+    it(`Charon main dataset: ${fromUrl} goes to ${toUrl} `, async () => {
+      const res = await fetch(charonUrl(fromUrl, false), {redirect: 'manual'});
+      expect(res.status).toEqual(307);
+      expect(decodeURIComponent(res.headers.get('location'))).toEqual(charonUrl(toUrl, false));
```

Comment:
_non-blocking_

These are only tests, so non-blocking, but this is the wrong way 'round in terms of URL encoding/decoding. The Location header is a URL, not a single URL component, so the more appropriate function is `decodeURI` not `decodeURIComponent`. Instead of decoding the Location URL at all though, the right way 'round fix is making `charonUrl` properly encode the URL it generates.